hist/hist/src/TGraph.cxx

```diff
@@ -2266,6 +2266,13 @@ void TGraph::SetHighlight(Bool_t set)
    painter->SetHighlight(this);
 }
 
+/// Set the histogram underlying the TGraph. This transfers the ownership of h to the TGraph. The preexisting fHistogram will be deleted.
+void  TGraph::SetHistogram(TH1F *h)
+{
+   delete fHistogram;
```

Comment:
Would it be useful to perhaps check if the pointer is not null before deleting?

Reply:
It's not needed since `delete nullptr;` is safe to call. (https://stackoverflow.com/questions/6731331/is-it-still-safe-to-delete-nullptr-in-c0x)